buildv2.py

```diff
@@ -0,0 +1,807 @@
+# them.
+GIT_VARIABLES_AND_COMMANDS = {
+    "git_commit": r"git log -n1 --format=%H",
+    "git_url": "git config --get remote.origin.url",
+    "git_branch": "git rev-parse --abbrev-ref HEAD",
+    "git_describe_v": r"git describe --match=v[34]\*",
+    "git_describe": "git describe"
+}
+
+# Modules tarball creation command.
+MODULES_TAR_CMD = "(cd {:s}; tar -Jcf {:s} lib/modules)"
+
+# Config merge script command.
+CONFIG_MERGE_CMD = (
+    "scripts/kconfig/merge_config.sh -O {0.kbuild_output:s} {1:s} {2:s} >"
+    " /dev/null 2>&1"
+)
+
+logger = logging.getLogger("kernelci-builder")
+handler = logging.StreamHandler()
+
+
+class BuildConfig(object):
+    """An object to hold build configuration parameters."""
+    def __init__(self):
+        # Not all of the configuration attributes are listed.
+        self.build_log = None
+        self.build_result = "PASS"
+        self.build_time = 0
+        self.debug = False
+        self.install = False
+        self.modules = False
+        self.publish = False
+        self.tree_name = None
+
+
+def check_proc_output(cmd, shell=True):
+    """Run a shell command in a subprocess and return its output."""
+    ret_val = None
+    try:
+        ret_val = subprocess.check_output(cmd, shell=shell)
+    except subprocess.CalledProcessError:
+        logger.error("Error running command: %s", cmd)
+        sys.exit(1)
+    return ret_val
+
+
+def do_make(build_cfg, target=None, log=False, cmd_vars=None):
+    """Execute the make command.
+
+    :param build_cfg: The BuildConfig object.
+    :type build_cfg: object
+    :param target: The target to make.
+    :type target: str
+    :param log: If to log the make output. Default to false.
+    :type log: bool
+    :param cmd_vars: List of variable names to add to the command line string.
+    :type cmd_vars: list
+    """
+    make_args = "-j{0.make_threads:d} -k".format(build_cfg)
+
+    if build_cfg.quiet:
+        make_args += " -s"
+
+    make_args += " ARCH={0.arch:s}".format(build_cfg)
+
+    if build_cfg.cross_compile:
+        make_args += " CROSS_COMPILE={0.cross_compile:s} ".format(build_cfg)
+
+    if build_cfg.ccache:
+        prefix = ""
+        if build_cfg.cross_compile:
+            prefix = build_cfg.cross_compile
+        make_args += " CC=\"ccache {:s}gcc\" ".format(prefix)
+
+    if build_cfg.kbuild_output:
+        make_args += " O={0.kbuild_output:s} ".format(build_cfg)
+
+    if target:
+        make_args += target
+
+    make_vars = ""
+    if cmd_vars:
+        def _create_var_string():
+            """Create the key=val string for the command line."""
+            for var in cmd_vars:
+                yield "{0:s}={1:s}".format(
+                    var.upper(), getattr(build_cfg, var))
+
+        make_vars = " ".join(_create_var_string())
+
+    make_cmd = "make {0:s} {1:s}".format(make_vars.strip(), make_args.strip())
+    logger.debug("Make command is: %s", make_cmd)
```

Comment:
The current build.py always prints the make_cmd, which is the better default, IMO, because when using interactively (which I do every day) it's good to have some output from the build script.  The current default gives no output at all.